packages/devextreme/js/__internal/ui/form/m_form.utils.ts

```diff
@@ -1,39 +1,57 @@
 import { extend } from '@js/core/utils/extend';
 import { isDefined } from '@js/core/utils/type';
+import type { GroupItem, SimpleItem, TabbedItem } from '@js/ui/form';
+import type { PreparedItem, TabItem } from '@ts/ui/form/m_form.items_runtime_info';
+import type { ConvertToLayoutManagerOptionsParams, LayoutManagerProperties } from '@ts/ui/form/m_form.layout_manager';
 
-export const createItemPathByIndex = (index, isTabs) => `${isTabs ? 'tabs' : 'items'}[${index}]`;
+export const createItemPathByIndex = (
+  index: number,
+  isTabs: boolean | undefined,
+): string => `${isTabs ? 'tabs' : 'items'}[${index}]`;
 
-export const concatPaths = (path1, path2) => {
+export const concatPaths = (
+  path1: string | undefined,
+  path2: string | undefined,
+): string | undefined => {
   if (isDefined(path1) && isDefined(path2)) {
     return `${path1}.${path2}`;
   }
+  // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
   return path1 || path2;
 };
 
-export const getTextWithoutSpaces = (text) => (text ? text.replace(/\s/g, '') : undefined);
+export const getTextWithoutSpaces = (text: string | undefined): string | undefined => (text ? text.replace(/\s/g, '') : undefined);
 
-export const isEqualToDataFieldOrNameOrTitleOrCaption = (item, fieldName) => {
+export const isEqualToDataFieldOrNameOrTitleOrCaption = (
+  item: PreparedItem,
+  fieldName: string,
+): boolean => {
   if (item) {
-    return item.dataField === fieldName
-            || item.name === fieldName
-            || getTextWithoutSpaces(item.title) === fieldName
-            || (item.itemType === 'group' && getTextWithoutSpaces(item.caption) === fieldName);
+    return (item as SimpleItem).dataField === fieldName
```

Comment:
```suggestion
    if (item) {
    const isSimpleItem = (item: PreparedItem): item is PreparedItem<SimpleItem> => 'dataField' in item;
    const isTabItem = (item: PreparedItem): item is PreparedItem<TabItem> => 'title' in item;
    const isGroupItem = (item: PreparedItem): item is PreparedItem<GroupItem> => item.itemType === 'group' && 'caption' in item;

    return (isSimpleItem(item) && item.dataField === fieldName)
      || item.name === fieldName
      || (isTabItem(item) && getTextWithoutSpaces(item.title) === fieldName)
      || (isGroupItem(item) && getTextWithoutSpaces(item.caption) === fieldName);
  }
```
here is some type issue with title in TabItem, but still can be used as a draft